### PR TITLE
[sorting] releases should be sorted by date, then by type (Geth, then Geth + tools) 

### PR DESCRIPTION
**Pull request by corwintines**
Adds sorting by type for releases

Fixes: https://github.com/ethereum/geth-website/issues/57

### CODE REVIEW

File: src/constants.ts [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import React from 'react';
 import { IconProps } from '@chakra-ui/react';
 import { WindowsLogo, MacosLogo, LinuxPenguin, SourceBranch } from './components/UI/icons';
+import exp from 'constants';
```
Comment (wackerow) [inline src/constants.ts:4]: Do we need this?
```suggestion
```

File: src/constants.ts [hunk 2]
```diff
@@ -194,6 +195,7 @@ export const ALL_MACOS_GETH_RELEASES_URL =
 export const ALL_MACOS_ALLTOOLS_GETH_RELEASES_URL =
   'https://gethstore.blob.core.windows.net/builds?restype=container&comp=list&prefix=geth-alltools-darwin';
 
+
```
Comment (wackerow) [inline src/constants.ts:198]:
```suggestion
```

**Review by wackerow — COMMENTED** [review]
Three tiny clean up suggestions, otherwise LGTM! Nice job

**Review by wackerow — APPROVED** [review]
Went ahead and applied those linting changes... looks good!

File: src/constants.ts (hunk 1 above)
Comment (corwintines) [reply src/constants.ts]: Nope, didnt realize I added that. Nice catch.

File: src/constants.ts [hunk 3]
```diff
@@ -216,3 +216,9 @@ export const RELEASE_NOTES_BASE_URL = 'https://github.com/ethereum/go-ethereum/r
 
 // Code snippet class constants
 export const CLASSNAME_PREFIX = 'language-';
+
+// Amount of releases per os constants
+export const AMOUNT_OF_RELEASES_PER_DARWIN = 2;
+export const AMOUNT_OF_RELEASES_PER_LINUX = 12;
```
Comment (nhsz) [inline src/constants.ts:222]: found that the amount of releases in linux is not always 12, in some older releases is 18 or 20 (check on https://geth.ethereum.org/downloads/).

between **1.10.9** and **1.9.25** included, is 18 (screenshot below)

Screen Shot 2022-12-14 at 09 43 54

**1.9.24** and older is 20 (screenshot below)

Screen Shot 2022-12-14 at 09 47 18

This only apply to Linux releases, in the other cases the amount is always the same fixed vlue.

File: src/pages/downloads.tsx [hunk 4]
```diff
@@ -195,36 +199,42 @@ export const getStaticProps: GetStaticProps = async () => {
...
           ALL_WINDOWS_DEV_BUILDS: getSortedReleases(
             WINDOWS_DEV_BUILDS_DATA,
-            WINDOWS_ALLTOOLS_DEV_BUILDS_DATA
+            WINDOWS_ALLTOOLS_DEV_BUILDS_DATA,
+            AMOUNT_OF_RELEASES_PER_WINDOWS
           ),
           // android
-          ALL_ANDROID_STABLE_RELEASES: getSortedReleases(ANDROID_STABLE_RELEASES_DATA),
-          ALL_ANDROID_DEV_BUILDS: getSortedReleases(ANDROID_DEV_BUILDS_DATA),
+          ALL_ANDROID_STABLE_RELEASES: getSortedReleases(ANDROID_STABLE_RELEASES_DATA, [], AMOUNT_OF_RELEASES_PER_MOBILE),
```
Comment (nhsz) [inline src/pages/downloads.tsx:233]: why the empty `[]` array for android and ios releases?

File: src/utils/getSortedReleases.ts [hunk 5]
```diff
@@ -1,6 +1,27 @@
 import { ReleaseData } from './../types';
 import { compareReleasesFn } from './compareReleasesFn';
 
-export const getSortedReleases = (releases: ReleaseData[], moreReleases: ReleaseData[] = []) => {
-  return releases.concat(moreReleases).sort(compareReleasesFn);
+export const getSortedReleases = (releases: ReleaseData[], moreReleases: ReleaseData[] = [], releaseChunkSize: number) => {
+  // function that takes an array and breaks it up into chunks of the specified size.
+  const chunkArray = (arr: ReleaseData[], releaseChunkSize: number) => {
```
Comment (nhsz) [inline src/utils/getSortedReleases.ts:6]: should be moved to a separate util

File: src/utils/getSortedReleases.ts [hunk 6]
```diff
@@ -1,6 +1,27 @@
...
+
+  const chunkedReleases = chunkArray(sortedReleasesByRelease, releaseChunkSize);
+  
+  return chunkedReleases.map((chunk) => {
+    return chunk.sort((a, b) => {
+      return a.arch.localeCompare(b.arch);
+    }).sort((a, b) => {
+      return b.release.label.localeCompare(a.release.label);
+    });
+  }).flat();;
```
Comment (nhsz) [inline src/utils/getSortedReleases.ts:26]: extra `;` here

**Review by nhsz — CHANGES_REQUESTED** [review]
Niceee, well done :D @corwintines 🙌

Found a minor issue for Linux releases that should be fixed.

File: src/pages/downloads.tsx (hunk 4 above)
Comment (corwintines) [reply src/pages/downloads.tsx:233]: Function takes 3 params, and one is an array, so passing empty array.

**Comment by corwintines** [comment]
Build error related to rate limiting from azure.

**Review by wackerow — APPROVED** [review]

**Review by wackerow — APPROVED** [review]
🎉

**Review by nhsz — APPROVED** [review]
Awesomeee 🎉🎉🎉, this was a hard one, thanks a lot for the refactor!

**Pull request merged**